vortex-datetime-parts/src/compute.rs

```diff
@@ -39,3 +54,193 @@ impl SliceFn for DateTimePartsArray {
         .into_array())
     }
 }
+
+impl ScalarAtFn for DateTimePartsArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        let DType::Extension(ext, nullability) = self.dtype().clone() else {
+            panic!("DateTimePartsArray must have extension dtype");
+        };
+
+        match ext.id().as_ref() {
+            LocalDateTimeArray::ID => {
+                let time_unit = try_parse_time_unit(&ext)?;
+                let divisor = match time_unit {
+                    TimeUnit::Ns => 1_000_000_000,
+                    TimeUnit::Us => 1_000_000,
+                    TimeUnit::Ms => 1_000,
+                    TimeUnit::S => 1,
+                };
+
+                let days = scalar_at(&self.days(), index)?.into_value();
```

Comment:
I think you will always get i64 here? If not we can switch over ptype and unwrap directly into the necessary type